### PR TITLE
#165038840 Edit user comment 

### PR DESCRIPTION
**Pull request by EfeAgare**
#### What does this PR do?

- It enables users to edit comment on articles
- Implements the feedback given by the PO during our demo.

#### Description of Task to be completed?
-  It enables users to edit comment on articles
- Make the user image and dropdown display on a new line in the mobile view

#### How should this be manually tested?
Clone the Repo, run `npm install`, then run `npm test`.

#### What are the relevant pivotal tracker stories?
[#165038840](https://www.pivotaltracker.com/story/show/165038840)
[#165030404](https://www.pivotaltracker.com/story/show/165030404)

#### Screenshots (if appropriate)
About to edit comment
Screen Shot 2019-04-03 at 7 03 49 PM
Screen Shot 2019-04-03 at 7 04 12 PM

Comment Edited
Screen Shot 2019-04-03 at 7 04 31 PM

NavBar

Screen Shot 2019-04-04 at 10 06 37 AM

Screen Shot 2019-04-04 at 10 04 35 AM

Screen Shot 2019-04-04 at 10 04 54 AM

### CODE REVIEW

File: src/components/common/Navbar.jsx [hunk 1]
```diff
@@ -20,6 +20,20 @@ export class NavBar extends Component {
...
   };
 
+  setUsername = (userData) => {
+    const { name, username } = userData;
+    let displayName;
+    if (name) {
+      const display = name.split(' ')[0];
+      displayName = display;
+    } else if (username) {
+      displayName = username.replace(/\d+/, '');
```
Comment (p-funky) [inline src/components/common/Navbar.jsx]: I believe the logic here is to remove the first set of numbers it finds in the string. Is it not possible for the username to be strictly numbers?

File: src/components/common/UserProfileCardComponent.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ const UserProfileCard = ({
 <Card style={{ width: '38rem', height: '33rem' }} className = 'p-5 user-card'>
 <div className = 'top-content mb-5'>
 <span className = 'float-left'>
-<p className = 'user-name'>{userName}</p>
+<p className = 'user-name'>{userName.replace(/\d+/, '')}</p>
```
Comment (p-funky) [inline src/components/common/UserProfileCardComponent.js]: I believe the logic here is to remove the first set of numbers it finds in the string. Is it not possible for the username to be strictly numbers?

File: src/components/containers/SingleArticle.jsx [hunk 3]
```diff
@@ -115,7 +114,10 @@ class SingleArticle extends Component {
             <div className="heading-info">
               <div className="article-info">
                 <div className="author-image-box">
-                  <p className="author"> written by {author.username}</p>
+                  <p className="author">
+                    {' '}
+                    written by {author.username ? author.username.replace(/\d+/, '') : ''}
+                  </p>
```
Comment (p-funky) [inline src/components/containers/SingleArticle.jsx:122]: I believe the logic here is to remove the first set of numbers it finds in the string. Is it not possible for the username to be strictly numbers?

File: src/styles/components/_commentItem.scss [hunk 4]
```diff
@@ -142,4 +146,19 @@
...
+  display: flex;
+  justify-content: flex-end;
+  .span {
+    margin-right: 20px;
+  }
+  .mr-4.edit-button {
+    margin-left: unset !important;
+    margin-right: unset !important;
+  }
+  .jXLEy {
```
Comment (p-funky) [inline src/styles/components/_commentItem.scss:161]: What classname is this?

Comment (EfeAgare) [reply src/styles/components/_commentItem.scss:161]: It's from the styled button component, I had to target it to align the cancel and submit button

Comment (p-funky) [reply src/styles/components/_commentItem.scss:161]: I'm aware it's from a styled component. My point is, doesn't that classname change? If yes, then you'd need to add your own static classname and style it.

Comment (EfeAgare) [reply src/styles/components/_commentItem.scss:161]: The classname does not change

**Review by divinediscipline — DISMISSED** [review]
Looks good to me

**Review by divinediscipline — DISMISSED** [review]

File: src/styles/components/_navbar.scss [hunk 5]
```diff
@@ -80,17 +91,38 @@ input.form-control:focus {
...
+    .show .input-group {
+      width: 100%;
     }
   }
+  .d-flex.flex-row.justify-content-between.nav-bar-name.navbar-nav {
+    justify-content: flex-end !important;
+    margin-right: 2.5rem;
+  }
+  .justify-content-between {
+    // -ms-flex-pack: justify!important;
```
Comment (primuse) [inline src/styles/components/_navbar.scss]: Please take out the commented line

**Review by omenkish — DISMISSED** [review]

**Review by omenkish — DISMISSED** [review]

**Review by ozone4real — APPROVED** [review]

**Review by primuse — APPROVED** [review]

**Pull request merged**